odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -331,38 +358,93 @@ namespace cockpit {
+
+                    // Rotate the model using DEG (m_rotation is in RAD!).
+                    glRotated(/*m_egoState.getRotation().getX()*180.0 / cartesian::Constants::PI*/0, 1, 0, 0);
+                    glRotated(/*m_egoState.getRotation().getY()*180.0 / cartesian::Constants::PI*/0, 0, 1, 0);
+                    // Rotate around z-axis and turn by 10 DEG.
+                    glRotated(/*m_egoState.getRotation().getZ()*/ (m_egoState.getRotation().getAngleXY() + M_PI/2.0)*180.0 / cartesian::Constants::PI + 13.0, 0, 0, 1);
+                    
+                    glPointSize(1.0f); //set point size to 1 pixel
+
+                    glBegin(GL_POINTS); //starts drawing of points
+                    glColor3f(1.0f,1.0f,0.0);//Yellow color
+
+                    uint16_t distance_integer(0);
+                    float xyDistance = 0.0f, xData = 0.0f, yData = 0.0f, zData = 0.0f;
+                    uint8_t intensity = 0;
+                    float azimuth = startAzimuth;
+                    for (uint32_t azimuthIndex = 0; azimuthIndex < numberOfAzimuths; azimuthIndex++) {
+                        float verticalAngle = START_V_ANGLE;
+                        for (uint8_t sensorIndex = 0; sensorIndex < entriesPerAzimuth; sensorIndex++) {
+                            sstr.read((char*)(&distance_integer), 2); // Read distance value from the string in a CPC container point by point
+                            float distance = 0.0;
+                            if (numberOfBitsForIntensity==0) {
+                                if(distanceEncoding == 0) {
+                                    distance = static_cast<float>(distance_integer / 100.0f); //convert to meter from resolution 1 cm
+                                } else {
+                                    distance = static_cast<float>(distance_integer / 500.0f); //convert to meter from resolution 2mm
                                 }
-                                azimuth+=azimuthIncrement;
+                                
+                                switch (distanceEncoding) {
+                                    case CompactPointCloud::CM : distance = static_cast<float>(distance_integer / 100.0f); //convert to meter from resolution 1 cm
+                                                                 break;
+                                    case CompactPointCloud::MM : distance = static_cast<float>(distance_integer / 500.0f); //convert to meter from resolution 2mm
+                                                                 break;
+                                    default : cout << "Wrong distance encoding! Must be either 0 or 1." << endl;
+                                              break;
+                                }       
+                            } else {
+                                uint16_t cappedDistance = (distance_integer & 0x03FFF) << numberOfBitsForIntensity;
```

Comment:
The 0x3FFF is stil unclear to me as this corresponds to cutting off the two top-most bits. Why are you using this constant?